hpedockerplugin/volume_manager.py

```diff
@@ -1397,19 +1397,11 @@ def _mount_volume(driver):
 
         pri_connection_info = None
         sec_connection_info = None
-        # Check if replication is configured
-        if self.tgt_bkend_config:
+        # Check if replication is configured and volume is
+        # populated with the RCG
+        if (self.tgt_bkend_config and 'rcg_info' in vol and
```

Comment:
There are some places where presence of self.tgt_bkend_config is being checked. E.g. in umount, following code is there:
        if self.tgt_bkend_config:
            _unmount_volume(self._remote_driver)
This flow tries to check presence of VLUN on the target array. If VLUN by the same name happens to be present there, this flow will undesirably unmount the volume thereby removing the VLUN entry. Please add a check to see if 'rcg_info' is present in the volume instance. Only then allow unmount here.

Please check other places where this condition is being checked and add 'rcg_info' condition to it wherever necessary.

e.g. _force_remove_vlun() has this check and requires to be guarded by presence of 'rcg_info'.

Reply:
Tested this condition after adding required checks